common/src/main/java/org/wso2/carbon/testgrid/common/Utils.java

```diff
@@ -38,31 +45,43 @@
      * @param command Command to execute
      * @return boolean for successful/unsuccessful command execution
      */
-    public static boolean executeCommand(String command, File workingDirectory) {
+    public static boolean executeCommand(String command, File workingDirectory) throws CommandExecutionException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Running shell command : " + command + ", from directory : " + workingDirectory.getName());
+        }
 
         ProcessBuilder processBuilder = new ProcessBuilder("/bin/bash", "-c", command);
         Process process;
+        BufferedReader reader = null;
         try {
-            if (workingDirectory !=null && workingDirectory.exists()) {
+            if (workingDirectory != null && workingDirectory.exists()) {
```

Comment:
We can use Java 8 try with resource statement here (refer [1]). Then we will not need the final block

[1] https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html